File: src/vegtrans.f
      subroutine vegtrans(x,y,a,b,numplt,numspc,numval)
c
c* passed in
c
	double precision x(numplt,numspc)
	double precision a(numval)
	double precision b(numval)
	integer numplt
	integer numspc
	integer numval
c
c* passed out
c
 	double precision y(numplt,numspc)
c
c********************* one **************************************
c
      do 10 i=1,numplt
        do 11 j=1,numspc
	if (x(i,j) .eq. 0.0) then
          y(i,j) = 0.0
        else
          do 12 k=1,numval
          if (x(i,j) .eq. a(k)) then
            y(i,j) = b(k)
            goto 11
          endif
   12     continue
        endif
   11   continue
   10 continue
c
      return
c
      end

File: R/vegtrans.R
vegtrans <- function (x,a,b)
{
    y <- matrix(-1,nrow=nrow(x),ncol=ncol(x))
    if (length(a) != length(b)) stop ("Transformation vectors must be the same length")
    if (!is.numeric(a) | !is.numeric(b) | !is.numeric(x)) stop
        ("All values must be numeric")
    numplt <- nrow(x)
    numspc <- ncol(x)
    numval <- length(a)
    tmp <-.Fortran("vegtrans",
        as.matrix(x),
        y=y,
        a,
        b,
        as.integer(numplt),
        as.integer(numspc),
        as.integer(numval),
        PACKAGE='labdsv')$y
    if (any(tmp<0)) {
        cat(paste("Value ",veg[tmp==-1],"not converted.  Set to NA\n"))
        tmp[tmp==-1] <- NaN
    }
    tmp <- as.data.frame(tmp)
    names(tmp) <- names(x)
    row.names(tmp) <- row.names(x)
    tmp
}

stdveg <- function (x,pltsum=NULL,spcmax=NULL) 
{
    if (!is.null(pltsum)) {
       tmp <- apply(x,1,sum)
       x <- sweep(x,1,tmp,"/")
       x <- sweep(x,1,pltsum,"*")
    }
    if (!is.null(spcmax)) {
       tmp <- apply(x,2,max)
       x <- sweep(x,2,tmp,"/")
    } 
    x
}

